apps/dapp-console-api/src/routes/faucet/FaucetRoute.ts

```diff
@@ -0,0 +1,470 @@
+  tx?: Hex
+  etherscanUrl?: string
+  error?: FaucetError
+  amountDistributed?: bigint
+  recipientAddress: Address
+  requestingWalletAddress?: Address
+  authMode: FaucetAuthMode
+}
+
+/**
+ * Route that handles requests for faucet funds.
+ */
+export class FaucetRoute extends Route {
+  private static readonly LOG_TAG = '[FaucetRoute]'
+  public readonly name = 'faucet' as const
+
+  public readonly faucetsInfo = 'faucetsInfo' as const
+  public readonly faucetsInfoController = this.trpc.procedure.query(
+    async () => {
+      return Promise.all(
+        this.faucets.map(async (faucet) => {
+          let isFaucetAvailable: boolean = true
+          try {
+            const faucetBalance = await faucet.getFaucetBalance()
+            // If the faucet balance has not been fetched and returns null
+            // assume that the faucet is available to claim from.
+            isFaucetAvailable =
+              faucetBalance === null || faucetBalance >= parseEther('1.0')
+          } catch (e) {
+            this.logger?.error(
+              `${FaucetRoute.LOG_TAG} Failed to fetch faucet balance for chain ${faucet.chainId}: ${e.message}`,
+              e,
+            )
+            metrics.faucetFetchBalanceFailures.inc({
+              chainId: faucet.chainId,
+            })
+          }
+          return {
+            displayName: faucet.displayName,
+            chainId: faucet.chainId,
+            isAvailable: isFaucetAvailable,
+            onChainDripAmount: faucet.onChainDripAmount,
+            offChainDripAmount: faucet.offChainDripAmount,
+          }
+        }),
+      )
+    },
+  )
+
+  public readonly nextDripsRoute = 'nextDrips' as const
+  /// Returns the seconds until the next drip can occur for each chain. If
+  /// no timeout exists for a chain then the value for that chain will be set to null.
+  public readonly nextDripsController = this.trpc.procedure
+    .use(isPrivyAuthed(this.trpc))
+    .input(
+      this.z.object({
+        chainId: this.z
+          .number()
+          .describe('The chains to check the last drip on'),
+        authMode: this.z
+          .enum(faucetAuthModes)
+          .describe('The authentication mode used for the drip'),
+        walletAddress: zodEthereumAddress.optional(),
+      }),
+    )
+    .query(async ({ ctx, input }) => {
+      const { chainId, authMode, walletAddress } = input
+      const userId = this.getUserIdForAuthMode(
+        ctx.session,
+        authMode,
+        walletAddress as Address | undefined,
+      )
+
+      const faucet = this.faucets.find((f) => f.chainId === chainId)
+
+      if (!faucet) {
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: `No faucet found for chainId ${chainId}`,
+        })
+      }
+
+      try {
+        const secondsUntilNextDrip = await faucet.secondsUntilNextDrip(
+          authMode,
+          userId as Hex,
+        )
+        return { chainId: faucet.chainId, secondsUntilNextDrip }
```

Comment:
now that all faucets on a specific network use the same l1 contract. we will probably want to refactor this to be by l1 network (ie sepolia) instead of by chain id